### PR TITLE
herman/ref-design-updates

### PR DESCRIPTION
**Pull request by hermanwikner**
### Description

1. Adds `TextWithTone` component to `@sanity/base`
2. Update `PaneItem` with `TextWithTone` to display published/draft status icons with time ago tool tips

Screenshot 2021-11-25 at 17 40 10

3. Update `ArrayItemReferenceInput` and `ReferenceInput` to set correct color on `TextWithTone` (published/draft status) in different states (selected and pressed)
4. Update `ReferencePreview` with `TextWithTone` component. The `__workaround_selected` prop could be removed here because of 3 👆.

### What to review

- Review the code in general
- Make sure that the code works as before
- Ensure that the status icons have the correct color in different situations, ie:
    - Published: Green when published, muted/dimmed when not
    - Draft: Yellow when there is a draft, muted/dimmed when not
    - Gets correct color when the option is being selected or pressed (see image below of example)
    
Screenshot 2021-11-25 at 17 56 25

Note: I had to use `any` in `ArrayItemReferenceInput` and `ReferenceInput` when passing the `EditReferenceLink` to the `as` prop.

### Notes for release
n/a

### CODE REVIEW

File: packages/@sanity/desk-tool/src/components/paneItem/PaneItem.tsx [hunk 1]
```diff
@@ -21,6 +22,18 @@ interface PaneItemProps {
   schemaType?: SchemaType
 }
 
+const PaneItemCard = styled(Card)`
+  /* TextWithTone uses its own logic to set color, and we therefore need 
+  to override this logic in order to set the correct color in different states */
+  &[data-selected],
+  &[data-pressed],
+  &:active {
+    [data-ui='TextWithTone'] {
```
Comment (bjoerge) [inline packages/@sanity/desk-tool/src/components/paneItem/PaneItem.tsx:31]: Any reason to prefer this over importing `TextWithTone` and using `${TextWithTone}` instead? That would make the dependency more explicit and easier to maintain, but might be benefits with this I'm not aware of.

File: packages/@sanity/form-builder/src/inputs/ReferenceInput/ArrayItemReferenceInput.tsx [hunk 2]
```diff
@@ -463,13 +476,12 @@ export const ArrayItemReferenceInput = forwardRef(function ReferenceInput(
...
               {hasRef ? (
-                <Card
+                <OptionCard
                   flex={1}
                   padding={1}
                   paddingRight={3}
                   radius={2}
-                  as={EditReferenceLink}
-                  //@ts-expect-error issue with styled components "as" polymorphism
+                  forwardedAs={EditReferenceLink as any}
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/ReferenceInput/ArrayItemReferenceInput.tsx:484]: Would be interesting to understand what the issue is with typescript here. Any ideas?

**Review by bjoerge — APPROVED** [review]
Looks good to me, great work @hermanwikner!

Would be good to get @mikolajdobrucki's approval on this as well.

**Review by mikolajdobrucki — APPROVED** [review]
Well done! LGTM 🔥 🔥 🔥 🔥 🔥 🔥

**Pull request merged**